compiler/rustc_hir_analysis/src/collect/predicates_of.rs

```diff
@@ -406,9 +402,9 @@ pub(super) fn explicit_predicates_of<'tcx>(
             // For a predicate from a where clause to become a bound on an
             // associated type:
             // * It must use the identity substs of the item.
-            //     * Since any generic parameters on the item are not in scope,
-            //       this means that the item is not a GAT, and its identity
-            //       substs are the same as the trait's.
+            //     * If generic parameters on the item are not in scope,
+            //       the item is not a GAT, and its identity substs are
+            //       the same as the trait's.
```

Comment:
And I think it means this but I'm not 100% sure.

Reply:
The previous wording makes more sense...

Since we're in the scope of the trait, and not in the scope of the associated type, we can't name any parameters of the GAT. That means that all we need to check are that the substs of the projection are the identity substs of the trait.

Reply:
Reworded more like how your comment looks like which I find a bit clearer https://github.com/rust-lang/rust/pull/105628/commits/b22769a7bdd40ba01bf0792c86ba763a525d8857